src/System.Data.Odbc/src/System/Data/Odbc/OdbcConnection.cs

```diff
@@ -561,7 +562,14 @@ internal void HandleError(OdbcHandle hrHandle, ODBC32.RetCode retcode)
 
         public override void Open()
         {
-            InnerConnection.OpenConnection(this, ConnectionFactory);
+            try
+            {
+                InnerConnection.OpenConnection(this, ConnectionFactory);
+            }
+            catch (DllNotFoundException e) when (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && e.Message.Contains(ExternDll.Odbc32))
+            {
+                throw new OdbcException($"Native dependencies missing. Odbc Object Driver missing. Follow instructions from http://www.unixodbc.org/ to install unixOdbc", new OdbcErrorCollection());
```

Comment:
I don't want to hard code in some website. What I recommend is 
```c#
throw new DllNotFoundException(SR.UnixOdbcNotFound, e);
```
So you preserve the exceptoin as an inner exceptoin in case it has more info (eg., it's not actually unixodbc missing)
The string would be eg "Could not find dependency. Please check that 'unixOdbc' is installed."

Reply:
Or maybe ... Please "Could not find dependency. Please check that 'unixOdbc' is installed, to provide 'libodbc'."